planner/core/logical_plan_builder.go

```diff
@@ -4160,9 +4160,15 @@ func (b *PlanBuilder) buildDataSource(ctx context.Context, tn *ast.TableName, as
 			result = us
 		} else {
 			go func() {
-				err := cachedTable.UpdateLockForRead(b.ctx, txn.StartTS())
-				if err != nil {
-					log.Warn("Update Lock Info Error")
+				defer func() {
+					if r := recover(); r != nil {
```

Comment:
some logging?

Reply:
Here will be re-designed in the following PRs.
Whether it success of fail do not affect the other part, so the log is useless and introduce misconception for the user.

I add this recover because if no recover here, the whole TiDB exit which is bad.
The actual cause of the panic is another bug of here: https://github.com/pingcap/tidb/blob/1b5b44047c6643e428798c332583d387256ba151/table/tables/cache.go#L62

This code assume the caller of `loadDataFromOriginalTable` inside a txn, if not, `sctx.Txn(true)` panic
I'll refactor the code and fix it in the following PRs, but for now, just recover and don't make TiDB panic

@lcwangchao

Reply:
upset on: @tiancaiamao L4149 shows b.ctx is already in a txn, why UpdateLockForRead is not?